src/Client.php

```diff
@@ -324,14 +324,16 @@ public function setUri($uri)
                 $this->clearAuth();
             }
 
+            $uri = $this->getUri();
+
             // Set auth if username and password has been specified in the uri
-            if ($this->getUri()->getUser() && $this->getUri()->getPassword()) {
-                $this->setAuth($this->getUri()->getUser(), $this->getUri()->getPassword());
+            if ($user = $uri->getUser() && $password = $uri->getPassword()) {
```

Comment:
We need brackets around expressions here, see following example - https://3v4l.org/oV5F5:
```php
<?php

function a() {
    return 1;
}

function b() {
    return 2;
}

if ($a = a() && $b = b()) {
    var_dump($a, $b);
}
```

results:
```
bool(true)
int(2)
```

and because of that I think there is missing proper test for that change.

Reply:
done ;)